blueprints/hr_assistant/general.py

```diff
@@ -0,0 +1,633 @@
+                responder.reply("I found the following details about {name}: {details}")
+                responder.frame = {}
+
+            else:
+                responder.reply("Hmmm, looks like this employee did not work here!\
+                                Would you like to know about someone else?")
+                responder.frame = {}
+        else:
+            responder.reply("I believe that person is not an employee here.")
+            responder.frame['info_visited'] = False
+
+
+@app.handle(intent='get_aggregate')
+def get_aggregate(request, responder):
+    """
+    When a user asks for a statistic, such as average, sum, count or percentage,
+    in addition to categorical filters (if any), this dialogue state captures all the
+    relevant entities, calculates the desired statistic function and returns it.
+
+    'function' entities represent the statistic functions - sum, average, percentage, count
+    """
+
+    # Checks for existing function entity from previous turn
+    func_entity = request.frame.get('function')
+
+    # If the user provides a new function entity, it replaces the one in context from previous turns
+    func_entities = [e for e in request.entities if e['type'] == 'function']
+    age_entities = [e for e in request.entities if e['type'] == 'age']
+
+    if func_entities:
+        func_entity = func_entities[0]
+
+    if func_entity:
+        # Resolve the recognized function entity to its canonical form, one that can be used in
+        # the output dialog as well as in the '_agg_function', which is used to calculate the values
+        # of these desired function entities
+        function, responder = _resolve_function_entity(responder, func_entity)
+
+        # If there are any categorical entities (eg. race, gender, department etc.) in the query
+        # that need filtering on, '_resolve_categorical_entities' fetches these entities, resolves
+        # them to their canonical form and filters the database on all the conditions requested by
+        # the user query
+        qa, size = _resolve_categorical_entities(request, responder)
+
+        # Handles Numerical Variables
+        if age_entities or _find_additional_age_entities(request, responder):
+            qa, size = _apply_age_filter(request, responder, qa, age_entities)
+            qa_out = qa.execute(size=size)
+
+            # Calculate and return desired mathemical value
+            responder.slots['value'] = _agg_function(qa_out, func=function, num_col='age')
+            responder.reply('Based on your query, the {function} is {value}')
+
+        # Functions count and percentage do not need to be specific to numerical features,
+        # unlike average and sum. For eg. 'how many males', require only the count of males
+        # and the total count. The following lines resolve all similar queries.
+        elif function not in ('avg', 'sum'):
+            qa_out = qa.execute(size=301)
```

Comment:
why 301?

Reply:
replaced that with a global var all across.